### PR TITLE
Improve getResources performance

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

**Comment by stuartwdouglas** [comment]
Previous failures should be fixed by the changes in JarClassPathElement

**Review by geoand — APPROVED** [review]
:+1:

**Comment by geoand** [comment]
Assuming CI passes, @gsmet we definitely want this one in for 1.5

**Comment by geoand** [comment]
Hm...

Weird errors:

```
2020-05-18T11:55:13.6450696Z [ERROR] COMPILATION ERROR : 
2020-05-18T11:55:13.6456050Z [INFO] -------------------------------------------------------------
2020-05-18T11:55:13.6464777Z [ERROR] error reading /home/runner/.m2/repository/net/bytebuddy/byte-buddy/1.10.5/byte-buddy-1.10.5.jar; zip END header not found
2020-05-18T11:55:13.6479210Z [ERROR] /home/runner/work/quarkus/quarkus/independent-projects/tools/common/src/test/java/io/quarkus/cli/commands/CreateProjectTest.java:[1,1] cannot access io.quarkus.cli.commands
2020-05-18T11:55:13.6481735Z   zip END header not found
2020-05-18T11:55:13.6493401Z [ERROR] /home/runner/work/quarkus/quarkus/independent-projects/tools/common/src/test/java/io/quarkus/maven/utilities/PomTransformerTest.java:[1,1] cannot access io.quarkus.maven.utilities
2020-05-18T11:55:13.6495858Z   zip END header not found
2020-05-18T11:55:13.6509928Z [ERROR] /home/runner/work/quarkus/quarkus/independent-projects/tools/common/src/test/java/io/quarkus/test/platform/descriptor/CombinedQuarkusPlatformDescriptorTest.java:[1,1] cannot access io.quarkus.test.platform.descriptor
```

**Comment by gsmet** [comment]
My opinion on this one is that it either goes into CR1 or it gets moved to 1.6.

So let's try to merge it in CR1.

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java [hunk 1]
```diff
@@ -179,10 +173,12 @@ public void reset(Map<String, byte[]> resources,
...
-        for (ClassPathElement i : elements) {
-            ClassPathResource res = i.getResource(nm);
-            if (res != null) {
-                resources.add(res.getUrl());
+        //TODO: in theory resources could have been added in dev mode
+        //but I don't thing this really matters for this code path
+        ClassPathElement[] providers = state.loadableResources.get(name);
+        if (providers != null) {
+            for (ClassPathElement element : providers) {
+                resources.add(element.getResource(nm).getUrl());
```
Comment (Postremus) [inline independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java:181]: I tried out my application with this pr. I get an NPE at this line now.
Drools 6.5.0.Final loads some resources.

Value of `nm`: `/META-INF/drools.default.packagebuilder.conf`
Value of `name`: `META-INF/drools.default.packagebuilder.conf`

element.getResource(nm) returns null.
element.getResource(name) returns a value.

Comment (gsmet) [reply independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java:181]: @stuartwdouglas I let you check that comment.

Comment (stuartwdouglas) [reply independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java]: There was a typo, some places were using the original name that did not have the / removed

**Pull request merged**